src/test/scala/uk/gov/nationalarchives/FileProcessorTest.scala

```diff
@@ -161,6 +161,25 @@ class FileProcessorTest extends AnyFlatSpec with MockitoSugar with TableDrivenPr
     res(s"$reference/TRE-$reference-metadata.json").checksum should equal("")
   }
 
+  "copyFilesFromDownloadToUploadBucket" should "return an error if the tar file contains a zero-byte file" in {
```

Comment:
If you move the logic out of the `copyFilesFromDownloadToUploadBucket` and put it into the `Lambda` class, then you can remove this test.